mitodl/micromasters
# Profile REST api honors privacy settings

#### What are the relevant tickets?
closes #387 

#### What's this PR do?
Makes sure that the profile rest view honors the profile privacy setting

#### Where should the reviewer start?
`profiles/views.py`, then `profiles/serializers.py`

#### How should this be manually tested?
You need 2 users (one with the `verified_micromaster_user` set to `True` and another with the same setting set to `False`) and a user with the profile filled out (let's call her `Foo`).
Then for each profile privacy setting of the Foo user, verify that the result of the profiles REST api works accordingly to what described in the story description.
Note you need to test also with the anonymous user.

#### What GIF best describes this PR or how it makes you feel?
![](http://i.giphy.com/SAeJELaeQikfu.gif)
